core/unix/loader.c

```diff
@@ -1123,6 +1123,29 @@ privload_relocate_rela(os_privmod_data_t *opd, ELF_RELA_TYPE *start, ELF_RELA_TY
         privload_relocate_symbol((ELF_REL_TYPE *)rela, opd, true);
 }
 
+/* XXX: This routine is called before dynamorio relocation when we are in a
+ * fragile state and thus no globals access or use of ASSERT/LOG/STATS!
+ */
+/* This routine is duplicated from module_relocate_relr for relocating dynamorio. */
+static void
+privload_relocate_relr(os_privmod_data_t *opd, const ELF_WORD *relr, size_t size)
+{
+    ELF_ADDR *r_addr = NULL;
+
+    for (; size; relr++, size -= sizeof(ELF_WORD)) {
```

Comment:
style: We use explicit bools so `size != 0;`

(Ditto on the other copy for all of these.)